.github/workflows/main.yml

```diff
@@ -162,6 +162,16 @@ jobs:
           ANVIL_FORK_URL: ${{ secrets.ANVIL_FORK_URL }}
           ANVIL_BLOCK_NUMBER: 15132000
       - name: Test
+        if: matrix.react == '18'
         run: pnpm test:coverage
         env:
           REACT_VERSION: ${{ matrix.react }}
+      - name: Test
+        if: matrix.react == '17'
+        run: |
+          cd packages/react
+          pnpm add -D react@17.0.2 react-dom@17.0.2
+          cd ../..
+          pnpm test:run react
```

Comment:
Vitest doesn't mock globals in the same way as Jest. So instead of using mocking React in a setup file, we'll install the target React 17 version here.